Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -180,26 +186,37 @@ def setup(self, dcfg=None, pstate=None, initialdlstatus=None, lm_network_engine_
 
                 self._logger.debug(u"setup: initialdlstatus %s %s", hexlify(self.tdef.get_infohash()), initialdlstatus)
 
-                self.create_engine_wrapper(lm_network_engine_wrapper_created_callback, pstate,
-                                           initialdlstatus=initialdlstatus, wrapperDelay=wrapperDelay)
+                def schedule_create_engine():
+                    if self.can_create_engine_wrapper():
+                        # Subroutine that can be scheduled by the taskManager as it cannot handle deferreds.
+                        def create_engine():
+                            self.cew_scheduled = True
+                            create_engine_wrapper_deferred = self.network_create_engine_wrapper(pstate, initialdlstatus)
+                            create_engine_wrapper_deferred.chainDeferred(deferred)
+
+                        # Schedule the call to create the engine with a delay defined by wrapperDelay
+                        self.session.lm.threadpool.add_task(create_engine, wrapperDelay)
+                    else:
+                        self.session.lm.threadpool.add_task(schedule_create_engine, 5)
+
+                schedule_create_engine()
 
             self.pstate_for_restart = pstate
+            return deferred
 
         except Exception as e:
             with self.dllock:
                 self.error = e
                 print_exc()
 
-    def create_engine_wrapper(self, lm_network_engine_wrapper_created_callback, pstate, initialdlstatus=None, wrapperDelay=0):
+    def can_create_engine_wrapper(self):
```

Comment:
If you want to make it nicer, you could have create_engine_wrapper return a deferred instead. This way you can move the periodic check here instead of having the rescheduled thing happening elsewhere, it would be neater.